x/emissions/keeper/inference_synthesis/synth_palette_weight.go

```diff
@@ -1,31 +1,51 @@
 package inference_synthesis
 
 import (
+	"fmt"
+
 	errorsmod "cosmossdk.io/errors"
+
 	alloraMath "github.com/allora-network/allora-chain/math"
 	emissions "github.com/allora-network/allora-chain/x/emissions/types"
 )
 
 // Given the current set of inferers and forecasters in the palette, calculate their
 // weights using the current regrets
 func (p *SynthPalette) CalcWeightsGivenWorkers() (RegretInformedWeights, error) {
-	// Calc std dev of regrets + epsilon
+	var regrets []alloraMath.Dec
+	infererRegrets := p.GetInfererRegretsSlice()
+	forecasterRegrets := p.GetForecasterRegretsSlice()
+	var err error
+	if len(infererRegrets) > 0 {
+		regrets = append(regrets, infererRegrets...)
+	}
+	if forecasterRegrets != nil && len(forecasterRegrets) > 0 {
+		regrets = append(regrets, forecasterRegrets...)
+	}
+	if len(regrets) == 0 {
+		return RegretInformedWeights{}, errorsmod.Wrapf(emissions.ErrEmptyArray, "No regrets to calculate weights")
+	}
+
+	fmt.Println("regrets :::", regrets)
+	// Calc std dev of regrets + f_tolerance
 	// σ(R_ijk) + ε
-	stdDevRegrets, err := alloraMath.StdDev(append(p.GetInfererRegretsSlice(), p.GetForecasterRegretsSlice()...))
+	stdDevRegrets, err := alloraMath.StdDev(regrets)
 	if err != nil {
 		return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error calculating standard deviation of regrets")
 	}
-	// Add epsilon to standard deviation
-	stdDevRegretsPlusEpsilon, err := stdDevRegrets.Abs().Add(p.Epsilon)
+	fmt.Println("stdDevRegrets :::", stdDevRegrets)
+	// Add f_tolerance to standard deviation
+	stdDevRegretsPlusFTolerance, err := stdDevRegrets.Abs().Add(p.FTolerance)
 	if err != nil {
-		return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error adding epsilon to standard deviation of regrets")
+		return RegretInformedWeights{}, errorsmod.Wrapf(err, "Error adding f_tolerance to standard deviation of regrets")
```

Comment:
what's the rule of thumb or f_tolerance vs epsilon?

Reply:
I think we use `fTolerance` for some calculations where it's explicit in the paper that we need it, represented by `ε`. Epsilon we use for some additional situations/potential edge cases where we need to avoid `0`.

`fTolerance = 0.01`
`Epsilon = 0.0001`

Reply:
that's a p good rule of thumb. if we're instituting that now, please ensure it's universally enforced by the time we merge my PR